engine/src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -197,25 +197,39 @@ trait CromwellApiService {
       case b: BodyPart => b.toStrict(duration).map(strict => b.name -> strict.entity.data)
     }.runFold(Map.empty[String, ByteString])((map, tuple) => map + tuple)
 
+    def toResponse(workflowId: WorkflowId): WorkflowSubmitResponse = {
+      WorkflowSubmitResponse(workflowId.toString, WorkflowSubmitted.toString)
+    }
+
+    def askSubmit(command: WorkflowStoreActor.WorkflowStoreActorSubmitCommand): Route = {
+      // NOTE: Asks were already here. Guessing this was a KISS port of per-request via:
```

Comment:
Hmm, do you have another link? I feel like that's only an example, as the section is titled "Longer Example", and not meant to announce that using `ask` is the new hotness. For example, in the same block of code, the actor being communicated with by the web route _is_ the database. I don't think flattening SOC is a new best practice either, and nowhere else on the page is `akka.pattern.ask` mentioned by the akka-http team.

I don't have a history of finding good links for scala via google, but here is an example of a "best practice" type blog post with discussion: [Don't use Actors for concurrency](https://www.chrisstucchio.com/blog/2013/actors_vs_futures.html)

I think we concur that the actor may be overkill. The code-comment is a heads up that one should NOT just copy-paste this code anymore without consideration.

If you have another link that says `ask` _should_ be used, and not _can_ be, I will use it.

I _do_ plan on returning someday to unwrap the actors off of these `someService.doFuture(request): Future[Response]` methods, but I unfortunately don't have the after-hours cycles for the next few months.

And you're 100% right that unwrapping an FSM is going to be a PITA. When I get to it, perhaps I'll push that FSM into an akka-stream `Graph` with composed sub-graphs... but I'll have to also cross that bridge later.